aggregation/utils.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2017 Hewlett Packard Enterprise Development LP
+//
+//    Licensed under the Apache License, Version 2.0 (the "License"); you may
+//    not use this file except in compliance with the License. You may obtain
+//    a copy of the License at
+//
+//         http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+//    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+//    License for the specific language governing permissions and limitations
+//    under the License.
+
+package aggregation
+
+func CheckSubArray(subArray []string, array []string) (isSubArray bool) {
+	// empty array is a sub array of all arraies
+	if len(subArray) == 0 {
```

Comment:
this function can be simplified, you should be able to assume true until you find something that doesn't match.

Reply:
yes, please let me know if the new pr makes more sense.